chainer/functions/normalization/l2_normalization.py

```diff
@@ -22,22 +23,25 @@ def check_type_forward(self, in_types):
         )
 
     def forward(self, inputs):
+        self.retain_inputs((0,))
         x, = inputs
         xp = cuda.get_array_module(x)
         norm = xp.linalg.norm(x, axis=self.axis) + self.eps
         norm = xp.expand_dims(norm, self.axis)
         return x / norm,
 
-    def backward(self, inputs, gy):
-        x = inputs[0]
-        gy = gy[0]
-        xp = cuda.get_array_module(x)
+    def backward(self, indexes, grad_outputs):
+        x, = self.get_retained_inputs()
+        gy, = grad_outputs
 
-        norm = xp.linalg.norm(x, axis=self.axis) + self.eps
-        norm = xp.expand_dims(norm, self.axis)
+        norm = chainer.functions.sqrt(chainer.functions.sum(
```

Comment:
How about assigning `F = chainer.functions` before these calculations? The code could be cleaner.

Reply:
Thank you, fixed as suggested.